storagext/lib/src/runtime/client.rs

```diff
@@ -98,26 +133,42 @@ impl Client {
         Call: subxt::tx::Payload,
         Keypair: subxt::tx::Signer<PolkaStorageConfig>,
     {
+        // Critical Section Start
+        let mut last_sent_nonce = self.last_sent_nonce.lock().await;
+        let current_nonce = self
+            .legacy_rpc
+            .system_account_next_index(&account_keypair.account_id())
+            .await?;
+        let current_header = self.legacy_rpc.chain_get_header(None).await?.unwrap();
+        let ext_params = DefaultExtrinsicParamsBuilder::new()
+            .mortal(&current_header, 8)
+            .nonce(current_nonce)
+            .build();
+
+        let submitted_extrinsic_hash = self
+            .client
+            .tx()
+            .create_signed_offline(call, account_keypair, ext_params)?
+            .submit()
+            .await?;
+
+        tracing::debug!(
+            "Previous nonce: {}, next nonce: {}",
+            last_sent_nonce,
+            current_nonce
+        );
+        *last_sent_nonce = current_nonce;
+        drop(last_sent_nonce);
+        // Critical Section End
```

Comment:
Using a scope instead of the explicit drop might make it clearer, especially since this has a "semantic scope" — i.e. the critical section. You can just return the value used afterwards — `submitted_extrinsic_hash`